faunadb-java-dsl/src/main/java/com/faunadb/client/query/Language.java

```diff
@@ -143,15 +143,6 @@ public Path at(int... others) {
   }
 
   /**
-   * Creates a new RefV literal value.
-   *
-   * @see <a href="https://fauna.com/documentation/queries#values">FaunaDB Values</a>
-   */
-  public static Expr Ref(String ref) {
```

Comment:
No because all refs must have a class, either a native class or an user class. Keeping that function we won't be able to specify a class, unless the parse the old format `classes/widget/12345`, but I don't think it will make sense.

Reply:
Let's say you're working at the expression level, where you can't construct refs with `new RefV(..)`, how would you create a new reference from a concatenated string? Or from a "sub select"? If creating a ref with a string is still a valid function call, the driver should keep it.

Reply:
If I understood your example correctly one could use the `"ref":"id":` function like: `Ref(Class(...), Concat(a, b))`

Reply:
I could selecting a value from a instance that contains `classes/users/123` and I want to turn it into a ref using `Ref(Var("my-var"))`. Is that still valid?

I seem to remember at least one customer storing foreign keys as strings for some reason.

Reply:
No, as far as I understood the spec this will not be possible anymore. Also that user should at least be storing the id part of the ref not the entire string like `classes/users/123`

Reply:
When are we removing support for `{ "@ref": "abc" }`? Shouldn't we just remove this constructor then?

I guess all I'm saying is that we should not restrict the query language until we have to. If `Ref("str")` still works, it should be there.

Reply:
as soon we activate the header `X-FaunaDB-API-Version: 2.1` (that means, when we merge this PR) that form will only be valid for native references, like `{"@ref": "classes"}` or `{"@ref": "indexes"}`

Reply:
ok, disregard the last comment, apparently core implemented support for thinks like